src/Microsoft.OData.Core/JsonLight/ODataJsonLightInputContext.cs

```diff
@@ -112,7 +112,7 @@ internal sealed class ODataJsonLightInputContext : ODataInputContext
             // don't know how to get MetadataDocumentUri uri here, messageReaderSettings do not have one
             // Uri metadataDocumentUri = messageReaderSettings..MetadataDocumentUri == null ? null : messageReaderSettings.MetadataDocumentUri.BaseUri;
             // the uri here is used here to create the FullMetadataLevel can pass null in
-            this.metadataLevel = JsonLightMetadataLevel.Create(messageInfo.MediaType, null, this.Model, this.ReadingResponse);
+            this.metadataLevel = JsonLightMetadataLevel.Create(messageInfo.MediaType, null, false, this.Model, this.ReadingResponse);

```

Comment:
>false,  [](start = 92, length = 7)

If a request returns @odata.type in the repsonse payload in none metadata, can we read it correctly with "false" setting? need a test case to cover it

Reply:
@xuzhg Could you point me to a similar test I could as a starting point for testing this change?

Reply:
For example a roundtrip test. in this test,
1) we write a payload with @odata.Type in None metadata level
2) we read this payload again.

---
In reply to: [539731622](https://github.com/OData/odata.net/pull/1946#discussion_r539731622) [](ancestors = 539731622)

Reply:
I don't think it is possible to round trip responses serialized with metadata=none as ODataReader does not support reading these types (see #1723). Given this, I don't see a good way to add a test for this.

Reply:
So far, I am ok.

---
In reply to: [541270448](https://github.com/OData/odata.net/pull/1946#discussion_r541270448) [](ancestors = 541270448)